packages/expo-notifications/ios/EXNotifications/Permissions/PermissionsModule.swift

```diff
@@ -0,0 +1,81 @@
+//  Copyright © 2024 650 Industries. All rights reserved.
+
+import ExpoModulesCore
+import UIKit
+import MachO
+
+public class PermissionsModule: Module {
+  var permissionsManager: (any EXPermissionsInterface)?
+  var requester: EXUserFacingNotificationsPermissionsRequester =
+    EXUserFacingNotificationsPermissionsRequester()
+
+  public func definition() -> ModuleDefinition {
+    Name("ExpoNotificationPermissionsModule")
+
+    OnCreate {
+      appContext?.permissions?.register([
+        requester
+      ])
+    }
+
+    AsyncFunction("getPermissionsAsync") { (promise: Promise) in
+      appContext?
+        .permissions?
+        .getPermissionUsingRequesterClass(
+          EXUserFacingNotificationsPermissionsRequester.self,
+          resolve: promise.resolver,
+          reject: promise.legacyRejecter
+        )
+    }
+
+    AsyncFunction("requestPermissionsAsync") { (requestedPermissions: NotificationPermissionRecord, promise: Promise) in
+      EXUserFacingNotificationsPermissionsRequester.setRequestedPermissions(requestedPermissions.toDictionary())
+      appContext?
+        .permissions?
+        .askForPermission(
+          usingRequesterClass: EXUserFacingNotificationsPermissionsRequester.self,
+          resolve: promise.resolver,
+          reject: promise.legacyRejecter
+        )
+    }
+  }
+}
+
+struct NotificationPermissionRecord: Record {
+  @Field
+  var allowAlert: Bool?
+  @Field
+  var allowBadge: Bool?
+  @Field
+  var allowSound: Bool?
+  @Field
+  var allowDisplayInCarPlay: Bool?
+  @Field
+  var allowCriticalAlerts: Bool?
+  @Field
+  var provideAppNotificationSettings: Bool?
+  @Field
+  var allowProvisional: Bool?
+
+  func count() -> Int {
+    return (self.allowAlert != nil ? 1 : 0) +
+    (self.allowBadge != nil ? 1 : 0) +
+    (self.allowSound != nil ? 1 : 0) +
+    (self.allowDisplayInCarPlay != nil ? 1 : 0) +
+    (self.allowCriticalAlerts != nil ? 1 : 0) +
+    (self.provideAppNotificationSettings != nil ? 1 : 0) +
+    (self.allowProvisional != nil ? 1 : 0)
+  }
+
+  func authorizationOptionValue() -> UNAuthorizationOptions {
+    var options: UNAuthorizationOptions = []
+    if self.allowAlert ?? false { options.insert(.alert) }
```

Comment:
`authorizationOptionValue ` also isn't used?


if `count` turns out not to be needed, then `false` could be the default value for the `Fields` of the `NotificationPermissionRecord`, making this a little less verbose.

Reply:
I'm now using this, and removed some Objective C code.